src/Toolkit/A.php

```diff
@@ -156,47 +156,65 @@ public static function join(array|string $value, string $separator = ', '): stri
 	/**
 	 * Merges arrays recursively
 	 *
-	 * @param int $mode Behavior for elements with numeric keys;
-	 *                  A::MERGE_APPEND:    elements are appended, keys are reset;
-	 *                  A::MERGE_OVERWRITE: elements are overwritten, keys are preserved
-	 *                  A::MERGE_REPLACE:   non-associative arrays are completely replaced
+	 * If last argument is an integer, it defines the
+	 * behavior for elements with numeric keys;
+	 * - A::MERGE_OVERWRITE:  elements are overwritten, keys are preserved
+	 * - A::MERGE_APPEND:     elements are appended, keys are reset;
+	 * - A::MERGE_REPLACE:    non-associative arrays are completely replaced
 	 */
-	public static function merge(array $array1, array $array2, int $mode = A::MERGE_APPEND): array
+	public static function merge(array|int ...$arrays): array
 	{
-		$merged = $array1;
+		// get mode from parameters
+		$last = A::last($arrays);
+		$mode = is_int($last) ? array_pop($arrays) : A::MERGE_APPEND;
+
```

Comment:
Because the method now accepts `array|int`, IMO we should ensure that `$arrays` only contains arrays at this point, otherwise the following code will break.

Reply:
Won't this be rather expensive for all the regular use cases just to throw an exception for the case when the code would break anyways?

Reply:
If we don't check, the method will break in unexpected ways and the resulting PHP warnings will be rather opaque.

Reply:
Still, I don't think such a performance-heavy check is warranted for the rare cases where someone passes two integers to an array method. I'd accept it to break in rather opaque ways in those rare cases instead of slowing down 99.9% of the cases.